Source/WebGPU/WGSL/EntryPointRewriter.cpp

```diff
@@ -162,8 +162,13 @@ void EntryPointRewriter::checkReturnType()
 
         auto& namedTypeName = downcast<AST::IdentifierExpression>(*maybeReturnType);
         if (auto* structType = std::get_if<Types::Struct>(namedTypeName.inferredType())) {
-            ASSERT(structType->structure.role() == AST::StructureRole::UserDefined);
+            if (m_stage == AST::StageAttribute::Stage::Fragment) {
+                if (structType->structure.role() == AST::StructureRole::UserDefined)
+                    structType->structure.role() = AST::StructureRole::FragmentOutput;
```

Comment:
I'm not sure if upgrading the role here is OK or if a new struct type needs to be constructed as in the `Stage::Vertex` case? I'll defer to @tadeuzagallo

Reply:
I am not sure either. I tried mirroring what the Vertex stage did, but it generated an additional struct type with the same members. 

Lets see what @tadeuzagallo thinks

Reply:
I think you'll need to duplicate the struct. The `UserDefined` struct should be used everywhere, except for the fragment stage output. In order to convert from the user defined struct to the output struct, check out MetalFunctionWriter.cpp:315, where we define an implicit constructor that takes the user-defined struct to create the output struct.

Otherwise, if you just change the role but the program also uses the struct elsewhere, e.g. in a helper function input or something like that, I think we'll get a compilation error, since those structs shouldn't have the same attributes on their fields.